### PR TITLE
Feature/us 449 rename betrokkene id fields

### PR DESCRIPTION
**Pull request by annashamray**
**Changes:**
1. Add `registratiedatum` to `Rol` model and API
1. Add support for dots and hyphens in `RolTypes` models

### CODE REVIEW

File: src/openapi.yaml [hunk 1]
```diff
@@ -6726,25 +6726,31 @@ components:
...
           maxLength: 1000
           minLength: 1
+        registratiedatum:
+          title: Registratiedatum
+          description: De datum waarop dit object is geregistreerd.
+          type: string
+          format: date-time
+          readOnly: true
       discriminator:
         propertyName: betrokkene_type
```
[the remark this answers is not in the document]
Comment (annashamray) [reply src/openapi.yaml]: Bugfix - https://github.com/VNG-Realisatie/vng-api-common/pull/61

[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openapi.yaml]: @annashamray `vng-api-common==0.55.1` has the fix

File: src/zrc/api/serializers.py [hunk 2]
```diff
@@ -45,8 +45,26 @@
 
 
 # serializers for betrokkene identificatie data used in Rol api
-class RolNatuurlijkPersoonSerializer(serializers.ModelSerializer):
+class RenameModelSerializer(serializers.ModelSerializer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # iterating through all fields in a nasty way to keep their order
+        fields_count = len(self.fields)
```
Comment (sergei-maertens) [inline src/zrc/api/serializers.py]: is `self.fields` not an `OrderedDict`?

Comment (annashamray) [reply src/zrc/api/serializers.py]: it's a `BindingDict` based on `OrderedDict`

File: src/openapi.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (annashamray) [reply src/openapi.yaml]: Thanks! A new version of `vng-api-common` is included to `ZRC`

File: src/zrc/api/serializers.py (hunk 2 above)
Comment (sergei-maertens) [reply src/zrc/api/serializers.py]: hrm, does this retain the order? I'd expect new keys to be added at the end, which is what happens when you do the rename. Or maybe it treats the non-rename as new key insertions and removes the key only to re-insert it at the end?

Comment (annashamray) [reply src/zrc/api/serializers.py]: during this loop I remove all the items from `self.fields` ([line](https://github.com/VNG-Realisatie/gemma-zaakregistratiecomponent/blob/d802250ff25aa52705d91b99392eee57e6005d77/src/zrc/api/serializers.py#L55))
and after that I add them again - with or without renaming
I couldn't make up a better way to change `self.fields` preserving the order

**Comment by sergei-maertens** [comment]
tests failing because of Docker shenanigans, wonderful

File: src/zrc/api/serializers.py (hunk 2 above)
Comment (sergei-maertens) [reply src/zrc/api/serializers.py]: yeah it doesn't look very attractive but it does the job and I can't think of a more elegant way myself

**Pull request merged**